doc/release/upcoming_changes/16650.compatibility.rst

```diff
@@ -0,0 +1,10 @@
+`np.genfromtxt` now correctly unpacks structured arrays
+-------------------------------------------------------------------------
+Previously, `genfromtxt` failed to unpack if it was called with ``unpack=True``
+and a structured datatype was inferred or passed to the ``dtype`` argument.
+
+Structured arrays will now correctly unpack into a list of arrays,
+one for each column::
+
+    >>> np.genfromtxt(StringIO("1 2.0 \n 3 4.0"), unpack=True)
+    [array([1, 3]), array([2., 4.])]]
```

Comment:
I think this example requires a structured `dtype=` being passed in?  It may be good to point/show an example which triggers the changed behaviour *without* passing in `dtype=` (if that is possible).

Reply:
(i.e. I tried the example on this branch, and the behaviour is not as written here – also there is an additional closing bracket)

Reply:
Sorry about that, it looks like the numbers in my example were too small so the inferred `dtype` was '<u4'. I replaced it with the test case for `dtype=None`.

Reply:
Maybe I have to take a closer look at the tests, or am I running the wrong code?  But it seems to me that this gives:
```
In [2]: from io import StringIO
In [3]: np.genfromtxt(StringIO("21 58.0 \n 35 72.0"), unpack=True)
Out[3]: 
array([[21., 35.],
       [58., 72.]])
```
OTOH, this one changed behaviour as expected:
```
In [5]: np.genfromtxt(StringIO("21 58.0 \n 35 72.0"), unpack=True, dtype="i,d")
```
so, right now I think the example is still incorrect, and if there are tests for this, they should probably be refined to check that the result is (correctly, I assume) a single array in the first case, since it should be a single array when there is no structured dtype detected (which seems to be the default?).

Unless this *should* change and I am missing something?

Reply:
Nope, you were running the right code; it turns out that the example was incorrect. I forgot that detection of a structured dtype relies on passing `dtype=None` to override the default `dtype=float` (otherwise, a single array is the correct return value). My example in the release note failed to specify the `dtype` argument. Should be correct now.